product_end_of_life/models/product.py

```diff
@@ -0,0 +1,187 @@
+        unique_variants = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1)
+        for template in unique_variants:
+            template.eol_date = template.product_variant_ids.eol_date
+        for template in (self - unique_variants):
+            template.eol_date = False
+
+    @api.multi
+    def _inverse_eol_date(self):
+        self.ensure_one()
+        if len(self.product_variant_ids) == 1:
+            self.product_variant_ids.eol_date = self.eol_date
+
+    @api.model
+    def email_product_eol_list(self,product_eol_ids, run_date):
+        if not product_eol_ids:
+           return
+        channel_id = self.env['mail.channel'].search([
+            ('name', '=', 'Product End-Of-Life Channel')],
+            limit=1)
+        data = "Run Date: %s" % str(run_date)
+        data += "<br/><br/><strong> [Product Name]    [Days Before Eol Date] \
+                    [End-of-Life Date] </strong>" 
+        for pid in product_eol_ids:
+           data += "<br/> [ %s ]" % pid.name
+           data += "[ %s ]" % str(datetime.strptime(pid.eol_date,\
+               "%Y-%m-%d").date() - run_date)
+           data += "[ %s ]" % str(pid.eol_date)
+        self.env['mail.message'].sudo(self._uid).create({
+            'model': 'mail.channel', 
+            'res_id': channel_id.id, 
+            'subject':'Product End-Of-Life Notification',
+            'body': data,
+            'message_type':'comment',
+            'subtype_id':1})
+
+    @api.model
+    def send_notify_product_eol(self):
+        # get the cronjob record
+        ir_act_server = self.env['ir.actions.server'].search([
+            ("code", "=", "model.send_notify_product_eol()")],
+            limit=1)
+        ir_cron = self.env['ir.cron'].search([
+            ('ir_actions_server_id', '=', ir_act_server.id)],
+            limit=1)
+        # set approaching eol interval
+        if ir_cron.approaching_type == 'weeks':
+            apr_day = ir_cron.approaching_number * 7
+            apr_month = 0
+        # TBD: need to correctly compute based on months.
+        elif ir_cron.approaching_type == 'months':
+            apr_day = ir_cron.approaching_number * 30
+            apr_month = 0
+        else:
+            apr_day = ir_cron.approaching_number
+            apr_month = 0
+        # set execute every interval
+        if ir_cron.interval_type == 'weeks':
+            int_day = ir_cron.interval_number * 7
+            int_month = 0
+        # TBD: need to correctly compute based on months.
+        elif ir_cron.interval_type == 'months':
+            int_day = ir_cron.approaching_number * 30
+            int_month = 0
+        elif ir_cron.interval_type == 'days':
+            int_day = ir_cron.interval_number
+            int_month = 0
+        # for minutes/hours throw an exception
+        else:
+            int_day = 1
+            int_month = 0
+        run_day = datetime.strptime(ir_cron.nextcall, \
```

Comment:
These should be

    run_day = datetime.strptime(ir_cron.nextcall,
                                "%Y-%m-%d %H:%M:%S").date()